synapse/storage/util/id_generators.py

```diff
@@ -446,23 +446,31 @@ def _load_current_ids(
 
             rows: List[Tuple[str, int]] = []
             for table, instance_column, id_column in tables:
+                logger.info(
+                    "table=%s instance_column=%s id_column=%s",
+                    table,
+                    instance_column,
+                    id_column,
+                )
                 sql = """
                     SELECT %(instance)s, %(id)s FROM %(table)s
                     WHERE ? %(cmp)s %(id)s
+                    /* Sort so that we handle rows in order for each instance. */
+                    ORDER BY %(id)s ASC
```

Comment:
Disregard this change for now.

Still trying to understand what we want to sort by and why, https://github.com/matrix-org/synapse/pull/13585#discussion_r951795379

Reply:
(Note we're in a loop so sorting in the SQL won't cause `rows` to be correctly sorted)